src/backend/distributed/planner/multi_planner.c

```diff
@@ -837,6 +874,27 @@ CurrentJoinRestrictionContext(void)
 
 
 /*
+ * CurrentRestrictionsMemoryContext returns the the last restriction's memory context
+ * from the list.
+ */
+static MemoryContext
+CurrentRestrictionsMemoryContext(void)
+{
+	PlannerRestrictionContext *plannerRestrictionContext = NULL;
+	MemoryContext restrictionsMemoryContext = NULL;
+
+	Assert(plannerRestrictionContextList != NIL);
+
+	plannerRestrictionContext =
+		(PlannerRestrictionContext *) linitial(plannerRestrictionContextList);
+
+	restrictionsMemoryContext = plannerRestrictionContext->restrictionsMemoryContext;
+
+	return restrictionsMemoryContext;
+}
```

Comment:
There is no specific reason, it seemed easier to read the code. 

I'm ok to remove those functions and use `CurrentRestrictionContext ()` as you specified.